### DISCUSSION TITLE
Firewall not working? Am I missing something?

### DISCUSSION THREAD

**Discussion by vertigo220** [celzero/rethink-app#835] [Q&A]
I tried blocking a game due to excessive video ads, but when I tap "Firewall" to turn the top row (Firewall, Unmetered, Metered) red, even though it says "This app is blocked," I keep getting ads, and when I check Rethink under the network log it shows a bunch of IPs that the app is/has connected to. If I block those, it just connects to different IPs instead, and continuing to block them is fruitless, since there are probably hundreds of different IPs for doubleclick.net, pagead2.googlesyndication.com, etc. So two questions:

1. Why is this not working, and it's still allowing the app to connect to these IPs despite being blocked?
2. Why doesn't it allow blocking by domains in addition to IP, as that would be far more useful?

This is on Android 13 on an unrooted phone (Samsung A71 5G), app version v053n (fdroid).

**Comment by ignoramous** [comment]
Hi there,

If you are comfortable doing so, can you share a screen recording of the rules, the network log, and the blocked app? It'll help see what's going wrong.

1. If *Network Log* shows the connections were blocked then they're indeed getting blocked by Rethink. Make sure you turn of *Allow Bypass* (check *Settings*). You can also optionally enable *Always-on VPN* and *Block connections without VPN* if you're on Android 10 or later ([ref](https://archive.is/XoKFk)) to prevent installed apps from connecting outside of the tunnel Rethink sets up. That said, bear in mind that *Google Play Framework* may download app-specific notifications (even for those apps that have been firewalled inside Rethink. This is because Rethink has no way to know which just which app *Google Play Framework* is downloading notifications for).
2. [`v054`](https://rethinkdns.com/download) (released 2 weeks ago) has per-app domain blocks.

**Comment by vertigo220** [comment]
The log showed they _weren't_ blocked, so I manually blocked them individually. I tried disabled Allow Bypass and I tried setting it as always-on and blocking connections without it and neither worked. I suspect it may be due to using the framework like you mentioned, which would be unfortunate. I'll try disabling Google Play to see if that works. I guess I also need to see why v054 isn't showing (I use Droid-ify, but it should still pull from the same repo). In the meantime, here's some screenshots:

![Screenshot_20230316_201106_RethinkDNS](https://user-images.githubusercontent.com/12024557/225794119-ea0c8757-3e92-4e8a-97b1-ce81e94927a7.jpg)
![Screenshot_20230316_215142_RethinkDNS](https://user-images.githubusercontent.com/12024557/225794132-890c43ef-baee-4046-9595-9ba575befed9.jpg)
![Screenshot_20230316_215157_RethinkDNS](https://user-images.githubusercontent.com/12024557/225794146-7bab80e8-75d3-4a99-98e4-abbe5d16af29.jpg)

LMK if you need anything else.
